# Description: Provide backwards compatibility for 01_gettext-desktopfiles.patch for X-{Debian,Ubuntu}-Gettext-Domain
# UbuntuSpecific: 01_gettext-desktopfiles.patch was changed to use X-GNOME-, so this is necessary until all our .desktop files are converted
Index: glib2.0-2.27.2/glib/gkeyfile.c
===================================================================
--- glib2.0-2.27.2.orig/glib/gkeyfile.c	2010-11-01 10:55:54.368605345 +0100
+++ glib2.0-2.27.2/glib/gkeyfile.c	2010-11-01 10:55:55.120609440 +0100
@@ -463,6 +463,16 @@
                                                     G_KEY_FILE_DESKTOP_GROUP,
                                                     G_KEY_FILE_DESKTOP_KEY_GETTEXT_DOMAIN,
                                                     NULL);
+  if (!key_file->gettext_domain)
+      key_file->gettext_domain = g_key_file_get_string (key_file,
+                                                        G_KEY_FILE_DESKTOP_GROUP,
+                                                        "X-Ubuntu-Gettext-Domain",
+                                                        NULL);
+  if (!key_file->gettext_domain)
+      key_file->gettext_domain = g_key_file_get_string (key_file,
+                                                        G_KEY_FILE_DESKTOP_GROUP,
+                                                        "X-Debian-Gettext-Domain",
+                                                        NULL);
 
   return TRUE;
 }
@@ -574,6 +584,16 @@
                                                     G_KEY_FILE_DESKTOP_GROUP,
                                                     G_KEY_FILE_DESKTOP_KEY_GETTEXT_DOMAIN,
                                                     NULL);
+  if (!key_file->gettext_domain)
+      key_file->gettext_domain = g_key_file_get_string (key_file,
+                                                        G_KEY_FILE_DESKTOP_GROUP,
+                                                        "X-Ubuntu-Gettext-Domain",
+                                                        NULL);
+  if (!key_file->gettext_domain)
+      key_file->gettext_domain = g_key_file_get_string (key_file,
+                                                        G_KEY_FILE_DESKTOP_GROUP,
+                                                        "X-Debian-Gettext-Domain",
+                                                        NULL);
 
   return TRUE;
 }
